lib/buffer-search.js

```diff
@@ -237,7 +237,7 @@ class BufferSearch {
       }
     }
 
-    this.emitter.emit('did-update', this.markers.slice());
+    if (changes.length) this.emitter.emit('did-update', this.markers.slice());
```

Comment:
Should this still be checking for `changes` then, or should it be checking for changed markers? (Not sure how to do that)